File: src/com/designpatterns/creational/abstractfactory/RoundedSquare.java
package com.designpatterns.creational.abstractfactory;

public class RoundedSquare implements IShape {

    public static final String TAG = "RoundedSquare";

    private final float side;
    private final float roundedEdge;

    public RoundedSquare(float side, float roundedEdge) {
        this.side = side;
        this.roundedEdge = roundedEdge;
    }

    @Override
    public float area() {
        return side * side;
    }

    @Override
    public float perimeter() {
        return 4 * side;
    }

    @Override
    public void draw() {
        System.out.printf(
                "Square:: area:%f, perimeter:%f, roundedEdge:%f%n",
                area(), perimeter(), roundedEdge);
    }
}

File: src/com/designpatterns/creational/abstractfactory/kotlin/RoundedSquare.kt
package com.designpatterns.creational.abstractfactory.kotlin

class RoundedSquare(private val side: Float, private val roundedEdge: Float) : IShape {

    override fun area(): Float {
        return side * side
    }

    override fun perimeter(): Float {
        return 4 * side
    }

    override fun draw() {
        System.out.printf("Square:: area:%f, perimeter:%f, roundedEdge:%f%n", area(), perimeter(), roundedEdge)
    }

    companion object {
        const val TAG = "RoundedSquare"
    }
}